### ISSUE TITLE
Punishment increased damage applied to player instead of enemy

### ISSUE THREAD

**Issue by Nimyron** [PathOfBuildingCommunity/PathOfBuilding#1888]
* **What happened?**
When Punishment is being used and "Is the enemy on low life ?" box is checked, punishment's "Cursed enemies take #% increased damage while on low life" is applied to the player instead of the enemy.
* **What were you trying to do?**
Using punishment on a build. The bug happens even in a new build with punishment as only skill.
* **What steps will reproduce the bug?**
Create new build, add punishment in the skills tab, check "Is the enemy on low life ?" in the configuration tab, check physical damage taken (by the player) in the calcs tab.
Also works on existing builds.
* **Does it reproduce every time?**
Yes, even across different builds, no matter if they're old or new.
* **Provide a build code:**
Cyclone berserker updated to the last version of PoB community : https://pastebin.com/aWie75i4
* **Screenshots:**
![2021-01-13_20h10_02](https://user-images.githubusercontent.com/45174799/104498924-a091ae00-55dc-11eb-862d-836e8cc30094.png)

NB : It has been reported before, in issue #1524 but the issue was closed and never solved it seems.

EDIT : New pastebin link with the punishment actually activated (sorry)

**Comment by benjaminysmall** [comment]
Wanted to comment with what I learned so far, although I don't have a fix yet:

- Punishment is the only skill in the game tagged as type "CurseBuff"
- This is because the _old_ version of Punishment (prior to the curse rework) gave the player a buff which increased player attack speed and damage, which scaled with Buff Effect mods.
- In `CalcPerform.lua`, there's a whole section of code (~15 lines) specifically dedicated to making this work. It has the comment `Curse applies a buff; scale by curse effect, then buff effect`
- This whole section shouldn't be needed anymore, because Punishment no longer provides the player a buff.
- All we need now is some way to say "If the enemy is on low life, apply an INC DamageTaken mod to it."
- I can't for the life of me figure out the proper syntax for that.

- Also, I tried looking around for other examples of similar stats, in order to learn the syntax, and I believe Divergent Physical to Lightning Support is also broken in a similar way. It is supposed to apply an INC PhysicalDamageTaken mod if enemy is shocked (by supported skill), and it does not appear to do that.

The PoB linked in the original issue doesn't actually use Punishment, so here are the test PoBs I've been using:
Punishment test: https://pastebin.com/eTSAVRLC
Divergent Phys to Lightning test: https://pastebin.com/GHgBSfdQ
